docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+spec:
+  # [!] Change isHostCluster to 'false' if you want to use a clusterRef and serviceAccountSecretRef
+  #     instead of using the mig-controller detected kubeconfig. Refer to mig-cluster-aws.yaml for an example.
+  isHostCluster: true
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+
+  Remote Mig cluster:
+
+  ```
+  apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-remote
+  namespace: openshift-config
+spec:
+  # [!] Change isHostCluster to 'true' if you'll be running the controller on this cluster.
+  #     Setting 'isHostCluster' to true will bypass using the clusterRef and serviceAccountSecretRef below.
+  isHostCluster: false
+
+  url: "https://my-remote-cluster-ip.nip.io:8443"
+
+  serviceAccountSecretRef:
+    name: sa-token-remote
+    namespace: openshift-config
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+- spec.azureResourceGroup: For azure clusters -- it's the resource group that in-cluster volumes use.
+
+- spec.caBundle: If the migcluster needs SSL verification for connections a user can supply a custom CA bundle. This field is required only when spec.Insecure is set false.
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
+
+- spec.serviceAccountSecretRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
+
+## MigHooks
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigHook
+metadata:
+  generateName: test-
+  name: test-btmv5
+  namespace: openshift-migration
+spec:
+  custom: false
+  image: 'quay.io/konveyor/hook-runner:latest'
+  playbook: >-
+    LSBuYW1lOiBTbGVlcCBmb3IgMTAgc2Vjb25kcyBhbmQgY29udGludWUgd2l0aCBwbGF5CiAgd2FpdF9mb3I6CiAgIadsfwerq0OiAxMA==
+  targetCluster: source
+```
+
+- spec.activeDeadlineSeconds: Specifies the highest amount of time for which the hook will run.
+
+- spec.custom: Specifies whether the hook is a custom Ansible playbook or a pre-built image. This is a required field.
+
+- spec.image: Specifies the image of the hook to be executed. This is a required field.
+
+- spec.playbook: Specifies the contents of the custom Ansible playbook in base64 format, it is used in conjunction with the custom boolean flag.
```

Comment:
@eriknelson @shubham-pampattiwar 

spec.playbook: What is the value of this parameter? Is the user supposed to copy the playbook here in base64?

Reply:
@apinnick The information on when the hook runs is stored in `migplan.spec.hook`

Reply:
@JaydipGabani  

What is spec.playbook? Is it the whole playbook, in base64 format?

Reply:
Yes, the entire playbook, base64 encoded. There is an example here with an encoded playbook: https://github.com/konveyor/mig-operator/blob/master/docs/usage/10.md#adding-a-mighook-cr